app/controllers/listings_controller.rb

```diff
@@ -488,7 +488,8 @@ def update
 
   def finalise_update(listing, shape, community, update_successful, old_availability)
     if update_successful
-      listing.location.update_attributes(params[:location]) if listing.location
+      location_params = permit_origin_loc_params(params.require(:listing))
```

Comment:
It seems that updating of the listing's `location` has been broken, because in the listing edit form there is no such field with `name="location"`, therefore `params[:location]` should have always returned `nil`